proxy/http3/Http3Transaction.cc

```diff
@@ -208,6 +206,7 @@ HQTransaction::transaction_done()
 {
   // TODO: start closing transaction
   super::transaction_done();
+  delete this;
```

Comment:
I am always scare about [this](https://isocpp.org/wiki/faq/freestore-mgmt#delete-this)

Reply:
I'm not a big fun of this pattern too, but this is probably the right place to delete/free/release transaction instance if we follow the design. H2 tries to free transactions here and there and it's a mess, but If you look at H1ClientTransaction, it's released here in `transaction_done()`.